src/Symfony/Component/HttpFoundation/Exception/SessionNotFoundException.php

```diff
@@ -18,7 +18,7 @@
  *
  * @author Jérémy Derussé <jeremy@derusse.com>
  */
-class SessionNotFoundException extends \LogicException implements RequestExceptionInterface
+class SessionNotFoundException extends \BadMethodCallException implements RequestExceptionInterface
```

Comment:
`BadMethodCallException` extends `BadFunctionCallException` which extends `LogicException`, so this change is OK. 

However, I don't like the fat that `SessionNotFoundException` extends `BadMethodCallException`, could we do it in another way? One that will make sure that `SessionNotFoundException` extends `LogicException` in Symfony 6?

Reply:
note: `SessionNotFoundException` has been added in 5.3, we can change the signature without taking care of BC.

I agree, `BadMethodCallException` is not the best choice, But I don't see what to do else for solving the original issue #40112 without breaking the BC.

I'm not sure we have tools to provide migration path on exceptions. ping @nicolas-grekas

Reply:
logic, bad method call .. in a way both work currently :)

strictly spoken i prefer extends Logic as well, esp. since the @throws BadMethodCallException isnt documented. So the BC issue is hypothetical to me.

Reply:
done